posthog/hogql/functions/mapping.py

```diff
@@ -139,6 +139,7 @@ class HogQLFunctionMeta:
     "not": HogQLFunctionMeta("not", 1, 1),
     # type conversions
     "toInt": HogQLFunctionMeta("toInt64OrNull", 1, 1),
+    "__toInt64": HogQLFunctionMeta("toInt64", 1, 1),  # TODO: `__` is supposed to mean this is raw, do we want to this?
```

Comment:
Personally, I think it's fine to have `toInt` mapped to `toInt64OrNull` and then all the other standard type conversions as is (e.g. `toInt64`, `toInt32`, etc). Because Clickhouse doesn't provide a generic `toInt`, we're therefore adding it in as a HogQL-defined func. Exposing native Clickhouse funcs are fine as is imo (e.g. dropping the double underscore)

Reply:
Agree with what @Gilbert09 said i.e. dropping the double underscore. Any opinions @mariusandra?

Reply:
Thanks for pointing out, I forgot about this TODO 😄 I like your idea of treating the size-aware functions as raw, and the general-purpose `toInt` as user-friendly.
However, we also have HogQL functions like `toUUID`, where adding `OrNull` is the _only_ transformation – if we ever need non-nullable UUID conversion, this pattern will not work there.
Also, as for _our_ users, I think it's worth reducing the surface area of the language for them, which is why I'm a bit hesitant about exposing _all_ the conversion functions to them (as there are dozens of combinations with ints, floats, and decimals of various sizes.

Now, I spent a couple hours trying to make HogQL function overloads to work in this context (column selected from a SELECT that's select from another SELECT) and this was painful with no good results. I don't think I understand this type system, or what a "type" even is.

Curious to hear from @mariusandra

Reply:
The type system work froze in https://github.com/PostHog/posthog/pull/17267 -- if it's ever finished, then we could see that `toTimeZone` doesn't change the type, and null handling will be automatic. Someone should put in the effort to finish it.

I don't mind exposing all the `toInt32`, etc methods. Sometimes `toInt8` is needed, and sometimes I need to go from an unsigned int to a signed int (`sign(sign())` works, but `toUInt64()` would be cleaner).

However with regards to null handling... it is very bad (consistent, but bad UX) in ClickHouse, so I'd like to improve this in HogQL... thus anything we can do automatically is good. Until the type system work is done, we can still manually wrap a lot of things with `assumeNotNull()`... 🤔

Reply:
`assumeNotNull()` doesn't really matter. It's that `-OrNull` variants only accept a string as an argument, and not other compatible types (e.g. `DateTime64` here).